apps/transport/lib/transport_web/controllers/aoms.ex

```diff
@@ -35,42 +37,48 @@ defmodule TransportWeb.AOMSController do
     }
   end
 
+  @spec csv(Plug.Conn.t(), map()) :: Plug.Conn.t()
   def csv(conn, _params) do
     conn
     |> put_resp_content_type("text/csv")
     |> put_resp_header("content-disposition", "attachment; filename=\"autorités_organisatrices_des_mobilités.csv\"")
     |> send_resp(200, csv_content())
   end
 
+  @spec aoms :: [map()]
   defp aoms do
     AOM
     |> preload([:datasets, :region, :parent_dataset])
     |> preload(datasets: :resources)
     |> join(:left, [aom], c in Commune, on: aom.insee_commune_principale == c.insee)
-    |> select([aom, commune], [aom, commune.nom])
+    |> select([aom, commune], {aom, commune.nom})
```

Comment:
did this work?

Reply:
yup, now it’s even typed! If you’d try to passe un tuple with 1 or 3 elements, it should fail